File: 0189-rotate-array/0189-rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        n = len(nums)
        k = k if k<n else k%n
        l = nums[n-k:] + nums[:n-k]
        for i in range(len(l)):
            nums[i] = l[i]
        

File: 0189-rotate-array/0189-rotate-array.cs

public class Solution {
    public void Rotate(int[] nums, int k) {
        int n = nums.Length;
        k = k>=n ? k%n : k;
        List<int> l = new List<int>();
        for(int i=n-k;i<n;i++)
        {
            l.Add(nums[i]);
        } 
        for(int i=0;i<n-k;i++)
        {
            l.Add(nums[i]);
        }
        for(int i=0;i<n;i++)
        {
            nums[i] = l[i];
        }
    }
}